tracking/tracking.go

```diff
@@ -0,0 +1,40 @@
+package tracking
+
+import (
+	"os"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/google/uuid"
+	rancher "github.com/rancher/go-rancher/v2"
+)
+
+const UUIDSetting = "install.uuid"
+
+var RancherUUID = loadRancherUUID()
+
+func loadRancherUUID() string {
+	client, err := rancher.NewRancherClient(&rancher.ClientOpts{
+		Url:       os.Getenv("CATTLE_URL"),
+		AccessKey: os.Getenv("CATTLE_ACCESS_KEY"),
+		SecretKey: os.Getenv("CATTLE_SECRET_KEY"),
+		Timeout:   5 * time.Second,
+	})
+	if err != nil {
+		log.Warn(err)
+		return uuid.Nil.String()
+	}
+
+	setting, err2 := client.Setting.ById(UUIDSetting)
+	if err2 != nil {
+		log.Warn(err2)
+		return uuid.Nil.String()
+	}
+
+	value, err3 := uuid.Parse(setting.Value)
```

Comment:
The primary reason for parsing the UUID is to detect if it is malformed (user messed with it). In that case, catalog-service will break for git.rancher.io repositories.

We could instead run a simple regexp to validate its appropriateness and cut out the vendor code.